More detailed error on failed installation of invalid feature

When running the Liberty plugin with a `server.xml` that contains an invalid feature, Liberty lists all features as invalid, and it's up to the user to see which one caused the problems.

Example:

```xml
<server>
    <featureManager>
        <feature>javaee-8.0</feature>
        <feature>mpHealthz-2.0</feature> <!-- misspelled -->
        ...
    </featureManager>
</server>
```

Causes:

```
[ERROR] Failed to execute goal net.wasdev.wlp.maven.plugins:liberty-maven-plugin:3.0.M1:dev (default-cli) on project openliberty-russia: com.ibm.ws.install.InstallException: CWWKF1299E: The following features could not be obtained: mpconfig-1.3 javaee-8.0 mphealthz-2.0. Ensure that the features are valid for Open Liberty. -> [Help 1]
```

Can that information be changed to only complain about the missing/invalid feature (`mpHealthz-2.0` in this example)?

Tested with `3.0.M1` and `Open Liberty 19.0.0.7/wlp-1.0.30.cl190720190711-1115`.

Cross-posted from OpenLiberty/ci.maven#493, as [this comment](https://github.com/OpenLiberty/ci.maven/issues/493#issuecomment-521391213) indicates that the message comes from Liberty.

--- a/dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/InstallKernelMap.java
+++ b/dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/InstallKernelMap.java
@@ -653,7 +653,7 @@ public class InstallKernelMap implements Map {
             data.put(ACTION_EXCEPTION_STACKTRACE, ExceptionUtils.stacktraceToString(e));
         } catch (RepositoryResolutionException e) {
             data.put(ACTION_RESULT, ERROR);
-            InstallException ie = ExceptionUtils.create(e, (Collection<String>) data.get(FEATURES_TO_RESOLVE), (File) data.get(RUNTIME_INSTALL_DIR), false, isOpenLiberty);
+            InstallException ie = ExceptionUtils.create(e, e.getTopLevelFeaturesNotResolved(), (File) data.get(RUNTIME_INSTALL_DIR), false, isOpenLiberty);
             data.put(ACTION_ERROR_MESSAGE, ie.toString());
             data.put(ACTION_EXCEPTION_STACKTRACE, ExceptionUtils.stacktraceToString(ie));
         } catch (InstallException e) {
@@ -797,7 +797,7 @@ public class InstallKernelMap implements Map {
     /**
      * Populate the feature name (short name if available, otherwise symbolic name) from the ESA's manifest into the shortNameMap.
      *
-     * @param esa ESA file
+     * @param esa          ESA file
      * @param shortNameMap Map to populate with keys being ESA canonical paths and values being feature names (short name or symbolic name)
      * @throws IOException If the ESA's canonical path cannot be resolved or the ESA cannot be read
      */
@@ -836,7 +836,7 @@ public class InstallKernelMap implements Map {
      * generate a JSON from provided individual ESA files
      *
      * @param generatedJson path
-     * @param shortNameMap contains features parsed from individual esa files
+     * @param shortNameMap  contains features parsed from individual esa files
      * @return singleJson file
      * @throws IOException
      * @throws RepositoryException

